### PR TITLE
Consume API

### PR DESCRIPTION
**Pull request by chriswhong**
This PR disables mirage and has the app consume data from `{api-domain}/projects/:projectid`.

`labs-zap-api` is now serving jsonapi compliant responses on `/projects` that are ready for this branch to consume.

### CODE REVIEW

File: app/templates/show-project.hbs [hunk 1]
```diff
@@ -16,73 +16,45 @@
 
         <div class="grid-x grid-padding-x">
           <div class="cell medium-6">
+            {{log model}}
```
Comment (allthesignals) [inline app/templates/show-project.hbs]: Plz remove these thanks

File: app/adapters/application.js [hunk 2]
```diff
@@ -9,4 +9,6 @@ export default class ApplicationAdapter extends JSONAPIAdapter {
 
     super._ajaxRequest(...arguments);
   }
+
+  host = 'http://localhost:3000'
```
Comment (allthesignals) [inline app/adapters/application.js]: This shouldn't be hard-coded

File: config/environment.js [hunk 3]
```diff
@@ -44,6 +44,9 @@ module.exports = function(environment) {
     // ENV.APP.LOG_TRANSITIONS = true;
     // ENV.APP.LOG_TRANSITIONS_INTERNAL = true;
     // ENV.APP.LOG_VIEW_LOOKUPS = true;
+    ENV['ember-cli-mirage'] = {
+      enabled: false
```
Comment (allthesignals) [inline config/environment.js]: This should be _enabled_ for development

**Comment by chriswhong** [comment]
All the issues above have been addressed, I added some environment logic to make sure the mirage API calls work in development and the real API gets called in production.

I had added a serializer that works for the real API but not for the mirage results, making this method in the application serializer dependent on environment did the trick.

```
  keyForAttribute(key) {
    return ENV.environment === 'production' ? key : dasherize(key);
  }
```

**Comment by allthesignals** [comment]
💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯

**Pull request merged**